src/app/shared/components/all-cards/allCards.controller.js

```diff
@@ -0,0 +1,48 @@
+export default class AllCardsController {
+    constructor(openWeatherMapsService, $timeout, routingFunctionsService, $state) {
+        this.openWeatherMapsService = openWeatherMapsService;
+        this.$timeout = $timeout;
+        this.routingFunctionsService = routingFunctionsService;
+        this.$state = $state;
+    }
+
+    $onInit() {
+        if (this.$state.params.city) {
+            this.cityParam = this.$state.params.city.toLowerCase();
+        }
+
+        this.openWeatherMapsService.getFiveDayForecastCity(this.cityParam)
+            .then((data) => {
+                this.todayForecast = [];
+                this.fiveDayForecast = new Map();
+
+                if (data && data.data && data.data.list) {
+                    data.data.list.forEach((hourlyForecast) => {
+                        const forecastDay = new Date(hourlyForecast.dt * 1000).getDate();
+                        if (forecastDay === new Date().getDate()) {
+                            this.todayForecast.push(hourlyForecast);
+                        } else {
+                            const data = this.fiveDayForecast.get(forecastDay) || [];
+                            data.push(hourlyForecast);
+                            this.fiveDayForecast.set(forecastDay, data);
+                        }
+                    });
+
+                    this.$timeout(() => this.fiveDayForecast = Array.from(this.fiveDayForecast.values()));
```

Comment:
I don't really get what are you doing here, why are you creating a map to at the end convert it to an array? All this seems really unnecessary.